src/include/OSL/oslexec.h

```diff
@@ -23,7 +23,9 @@ struct PerThreadInfo;
 class ShadingContext;
 class ShaderSymbol;
 class OSLQuery;
+#ifdef OSL_USE_BATCHED
```

Comment:
This may not work the way you think. This header, oslexec.h, is read by client/app code, but OSL_USE_BATCHED is only defined by the OSL build system itself. So these definitions won't be visible to a renderer if they don't know to enable the preprocessor symbol.

Instead, what I would suggest is that in oslconfig.h.in, you add:

    #cmakedefine OSL_USE_BATCHED

And then that symbol will be defined (in the generated `oslconfig.h`) if the CMake variable OSL_USE_BATCHED is defined and nonzero, otherwise will omit the definition. (See https://cmake.org/cmake/help/latest/command/configure_file.html) Then the rest of the `#ifdef OSL_USE_BATCHED` will work as intended, even in client code that doesn't use the OSL build system -- it will appear defined in oslconfig.h any time that the OSL build itself enabled batched shading.